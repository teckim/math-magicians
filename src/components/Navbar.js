// import { Link } from 'react-router-dom';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import IconButton from '@mui/material/IconButton';
import MenuIcon from '@mui/icons-material/Menu';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';
import Container from '@mui/material/Container';

const navItems = [
  { title: 'Home', link: '/' },
  { title: 'Calculator', link: '/calculator' },
  { title: 'Quotes', link: '/quotes' },
];

const Navbar = () => (
  <AppBar component="nav" position="relative">
    <Container>
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="open drawer"
          edge="start"
          sx={{ mr: 2, display: { sm: 'none' } }}
        >
          <MenuIcon />
        </IconButton>
        <Typography
          variant="h6"
          component="div"
          sx={{ flexGrow: 1, display: { xs: 'none', sm: 'block' } }}
        >
          Math Magicians
        </Typography>
        <Box sx={{ display: { xs: 'none', sm: 'block' } }}>
          {navItems.map(({ title, link }) => (
            <Button key={title} href={link} sx={{ color: '#fff' }}>
              {title}
            </Button>
          ))}
        </Box>
      </Toolbar>
    </Container>
  </AppBar>
);

export default Navbar;
