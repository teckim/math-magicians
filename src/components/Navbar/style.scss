.navbar {
  max-width: 960px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 0 auto;

  &__title {
    font-size: 20px;
  }

  &__links {
    display: flex;
    align-items: center;
    list-style: none;
    padding: 0;
    margin: 0;

    &-item {
      padding: 0 12px;

      &:not(:last-child) {
        margin-right: 8px;
      }

      & > a {
        text-decoration: none;
        color: #000;
      }
    }
  }
}
