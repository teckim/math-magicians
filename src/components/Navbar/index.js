import { Link } from 'react-router-dom';

import './style.scss';

const Navbar = () => (
  <nav className="navbar">
    <h2 className="navbar__title">Math Magicians</h2>
    <ul className="navbar__links">
      <li className="navbar__links-item">
        <Link to="/">Home</Link>
      </li>
      <li className="navbar__links-item">
        <Link to="/calculator">Calculator</Link>
      </li>
      <li className="navbar__links-item">
        <Link to="/quotes">Quotes</Link>
      </li>
    </ul>
  </nav>
);

export default Navbar;
