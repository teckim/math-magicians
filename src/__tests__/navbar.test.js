import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

describe('navbar component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<BrowserRouter><Navbar /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
