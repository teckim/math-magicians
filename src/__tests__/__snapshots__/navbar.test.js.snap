// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar component should render correctly 1`] = `
<nav
  className="navbar"
>
  <h2
    className="navbar__title"
  >
    Math Magicians
  </h2>
  <ul
    className="navbar__links"
  >
    <li
      className="navbar__links-item"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="navbar__links-item"
    >
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="navbar__links-item"
    >
      <a
        href="/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
